js/controller.js

```diff
@@ -869,8 +870,22 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
      * @param multiAudio {Object} - audio which fetched for the current video
      * @param multiAudio.tracks {Array} - list of objects with data for each audio
      */
-    onMultiAudioFetched: function(event, multiAudio) {
-      if (this.state.showMultiAudioIcon) { //if param showMultiAudioIcon is set to true
+    onMultiAudioFetched: function (event, multiAudio) {
+      if (this.state.showMultiAudioIcon) { // if param showMultiAudioIcon is set to true
+        this.state.multiAudio = multiAudio;
+        this.renderSkin();
+      }
+    },
+
+    /**
+     * The function is called when event MULTI_AUDIO_CHANGED was caught;
+     * The function sets value for this.state.multiAudio
+     * @param event {String} name of a event
+     * @param multiAudio {Object} - audio which fetched for the current video
+     * @param multiAudio.tracks {Array} - list of objects with data for each audio
+     */
+    onMultiAudioChanged: function (event, multiAudio) {
```

Comment:
Will onMultiAudioChanged and onMultiAudioFetched always do the same thing? If so, it might be better for them to call a single function, or have one function call the other (for example, inside the onMultiAudioFetched callback, we just call onMultiAudioChanged). This makes it easier to maintain as we don't have to maintain duplicate code across two callbacks.

Reply:
Good point! Yes, for now they always do the same thing, but the events they react to are different, so they might start doing different things later and then we'll have to deal with premature generalisation.